mne/preprocessing/ica.py

```diff
@@ -144,36 +161,57 @@ class ICA(ContainsMixin):
     Parameters
     ----------
     n_components : int | float | None
-        The number of components used for ICA decomposition. If int, it must be
-        smaller than `max_pca_components`. If None, all PCA components will
-        be used. If float between 0 and 1, components will be selected by the
-        cumulative percentage of explained variance.
+        Controls the number of PCA components from the pre-ICA PCA entering the
+        ICA decomposition in the `fit()` method.
+        If None (default), all PCA components will be used
+        (== `max_pca_components`).
+        If int, must be <= `max_pca_components`.
+        If float between 0 and 1, the number of components selected matches the
+        number of components with a cumulative explained variance below
+        `n_components` (<1> resulting in `max_pca_components`).
+        The actual number of components resulting from evaluating this
+        parameter in the `fit()` method is stored in the attribute
+        `n_components_`.
+        (Note that the trailing ``_`` signifies attributes added to the object
+        by methods to store results from the actual calculations.)
```

Comment:
Super long docstrings hurt readability. Let's move this `_`-related note to the Notes section. It's a general point anyway.